lib/synapse-plugins/window-switcher-plugin.vala

```diff
@@ -0,0 +1,80 @@
+/*
+* Copyright (c) 2019 Ranfdev <ranfdev@gmail.com>
+*               2019 elementary LLC.
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Ranfdev <ranfdev@gmail.com>
+*/
+
+namespace Synapse {
+    public class WindowSwitcherPlugin: Object, Activatable, ItemProvider {
+        public bool enabled { get; set; default = true; }
+
+        public void activate () { }
+
+        public void deactivate () { }
+
+        static void register_plugin () {
+            DataSink.PluginRegistry.get_default ().register_plugin (typeof (WindowSwitcherPlugin),
+                                                                    "Window switcher",
+                                                                    _("Search and open active windows"),
+                                                                    "switcher",
+                                                                    register_plugin);
+        }
+
+        static construct {
+            register_plugin ();
+        }
+
+
+        public signal void load_complete ();
+
+        public class WindowMatch: Object, Match {
+          public string title { get; construct set; }
```

Comment:
I need some help...

I need to add a property here:
`public Wnck.Window window;`

but by doing this i get some compile errors:
```
lib/synapse-plugins/synapse-plugins.vapi:64.11-64.14: error: The symbol `Wnck' could not be found
                        public Wnck.Window window;
```
I've already added libwnck-3.0 as a dependency, but it doesn't work...

Reply:
Have you tried rebuilding the entire project?

I will also tell you that we won't likely accept this since it uses libwnck and we don't want to introduce dependencies that require X to work.

Reply:
Yes, i've deleted the entire build folder and rebuilt everything from scratch. Same problem...

Ah... that's a problem because i don't know how to list open windows without libwnck. Maybe i could interface with gala, but that seems difficult